ci/dev/vscode.patch

```diff
@@ -689,6 +689,53 @@ index 18d3d04fd20335975293e37b3b641120dd92da20..4e49f9d63623da6c84624144765f76ec
  			(err: any, socket: ISocket | undefined) => {
  				if (err || !socket) {
  					options.logService.error(`${logPrefix} socketFactory.connect() failed. Error:`);
+@@ -411,16 +411,23 @@ abstract class PersistentConnection extends Disposable {
+ 		}
+ 		const logPrefix = commonLogPrefix(this._connectionType, this.reconnectionToken, true);
+ 		this._options.logService.info(`${logPrefix} starting reconnecting loop. You can get more information with the trace log level.`);
+-		this._onDidStateChange.fire(new ConnectionLostEvent());
```

Comment:
Only concern I have is whether it's ok to not fire this here. Maybe it prevents VS Code from trying to do things on the connection?

But I'm not sure. Just something to note. cc @code-asher

Reply:
@nhooyr @mgmachado @code-asher 
a search in the vscode repo indicates 3 subscribers to `PersistentConnectionEventType`:
- `RemoteAgentConnectionStatusListener` - [github link](https://github.com/microsoft/vscode/blob/f3a23697a90f49276cdb4f5531e931f9d6134086/src/vs/workbench/contrib/remote/browser/remote.ts#L770-L821)
- `RemoteExtensionHostEnvironmentUpdater` - [github link](https://github.com/microsoft/vscode/blob/8fcf778ba251131780f2a110d311161f39b97b03/src/vs/workbench/contrib/remote/electron-sandbox/remote.contribution.ts#L82-L91)
- `ExtensionService` - [github link](https://github.com/microsoft/vscode/blob/522391c3b5a04e2b31ca2a03d31658b79c4b8ba3/src/vs/workbench/services/extensions/electron-browser/extensionService.ts#L317-L324)

The `ExtensionService` does act on the `ConnectionLost` event - so that it can `_clearResolvedAuthority`. The `RemoteExtensionHostEnvironmentUpdater` is only interested in the `ConnectionGain` event - also related to the remote authority.

An alternative approach could be to modify `RemoteAgentConnectionStatusListener` to handle the events differently. In the same file, it seems like changing [VisibleProgress](https://github.com/microsoft/vscode/blob/f3a23697a90f49276cdb4f5531e931f9d6134086/src/vs/workbench/contrib/remote/browser/remote.ts#L609-L675) with some sort of delayed activation might be cleaner (after a name change perhaps).